src/main/java/com/tileman/TilemanPluginPanel.java

```diff
@@ -0,0 +1,440 @@
+                }
+
+                gameModeSelectPanel.add(gameModeDropdownPanel);
+                gameRulesPanel.add(gameModeSelectPanel);
+            }
+
+
+            {
+                JCollapsePanel customGameModeCollapsable = new JCollapsePanel("Custom Game Mode", gameModeOpen, (Boolean state) -> this.gameModeOpen = state);
+                customGameModeCollapsable.setBorder(BorderFactory.createLineBorder(Color.black));
+                customGameModeCollapsable.setInnerLayout(new BorderLayout());
+
+                gameRulesPanel.add(customGameModeCollapsable);
+
+                {
+                    JCheckBox customGameMode = new JCheckBox("Enable Custom Game Mode");
+                    customGameModeCollapsable.add(customGameMode, BorderLayout.NORTH);
+
+                    customGameMode.setAlignmentX(CENTER_ALIGNMENT);
+                    customGameMode.setSelected(profileManager.isEnableCustomGameMode());
+                    customGameMode.addActionListener(l ->  {
+                        profileManager.setEnableCustomGameMode(customGameMode.isSelected());
+                        rebuild();
+                    });
+                }
+
+                {
+                    JPanel rulesPanel = new JPanel();
+                    addVerticalLayout(rulesPanel);
+                    callbacks.add(() -> setJComponentEnabled(rulesPanel, profileManager.isEnableCustomGameMode()));
+
+                    customGameModeCollapsable.add(rulesPanel, BorderLayout.CENTER);
+
+                    {
+                        JCheckBox allowTileDeficit = new JCheckBox("Allow Tile Deficit");
+                        allowTileDeficit.setAlignmentX(CENTER_ALIGNMENT);
+                        allowTileDeficit.setSelected(profileManager.isAllowTileDeficit());
+                        allowTileDeficit.addActionListener(l -> profileManager.setAllowTileDeficit(allowTileDeficit.isSelected()));
+                        rulesPanel.add(allowTileDeficit);
+                    }
+
+                    {
+                        JCheckBox tilesFromLevels = new JCheckBox("Tiles From Levels");
+                        tilesFromLevels.setAlignmentX(CENTER_ALIGNMENT);
+                        tilesFromLevels.setSelected(profileManager.isTilesFromTotalLevel());
+                        tilesFromLevels.addActionListener(l -> profileManager.setTilesFromTotalLevel(tilesFromLevels.isSelected()));
+                        rulesPanel.add(tilesFromLevels);
+                    }
+
+                    {
+                        JCheckBox tilesFromExp = new JCheckBox("Tiles From Exp");
+                        tilesFromExp.setAlignmentX(CENTER_ALIGNMENT);
+                        tilesFromExp.setSelected(profileManager.isTilesFromExp());
+                        tilesFromExp.addActionListener(l -> profileManager.setTilesFromExp(tilesFromExp.isSelected()));
+                        rulesPanel.add(tilesFromExp);
```

Comment:
Could be worth adding a function/factory for creating these checkboxes.

Reply:
Done. Should clean up the style repetition

Reply:
Very clean now :)